From kristen.c.accardi@intel.com Thu Jul  5 11:11:11 2007
From: Gary Hade <garyhade@us.ibm.com>
Date: Thu, 05 Jul 2007 11:10:47 -0700
Subject: PCI: hotplug: acpiphp: remove hot plug parameter write to PCI host bridge
To: gregkh@suse.de
Cc: Gary Hade <garyhade@us.ibm.com>, lcm@us.ibm.com, Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Message-ID: <20070705181110.952652964@intel.com>


From: Gary Hade <garyhade@us.ibm.com>

acpiphp is writing hot plug parameters to the PCI host bridge
PCI config space.  This patch removes the incorrect operation.

Signed-off-by: Gary Hade <garyhade@us.ibm.com>
Cc: <lcm@us.ibm.com>
Signed-off-by: Kristen Carlson Accardi <kristen.c.accardi@intel.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---

---
 drivers/pci/hotplug/acpiphp_glue.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/pci/hotplug/acpiphp_glue.c
+++ b/drivers/pci/hotplug/acpiphp_glue.c
@@ -1368,6 +1368,9 @@ static void program_hpp(struct pci_dev *
 			(dev->class >> 8) == PCI_CLASS_BRIDGE_PCI)))
 		return;
 
+	if ((dev->class >> 8) == PCI_CLASS_BRIDGE_HOST)
+		return;
+
 	pci_write_config_byte(dev, PCI_CACHE_LINE_SIZE,
 			bridge->hpp.t0->cache_line_size);
 	pci_write_config_byte(dev, PCI_LATENCY_TIMER,
